File: Python/JSON Payload/pdf-with-added-image.py
import requests
import json

with open('/path/to/pdf_file', 'rb') as f:
    upload_pdf_data = f.read()

print("Uploading PDF file...")
upload_pdf_response = requests.post(url='https://api.pdfrest.com/upload',
                    data=upload_pdf_data,
                    headers={'Content-Type': 'application/octet-stream', 'content-filename': 'pdf_file.pdf', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})

print("Upload PDF response status code: " + str(upload_pdf_response.status_code))


with open('/path/to/image_file', 'rb') as f:
    upload_image_data = f.read()

print("Uploading image file...")
upload_image_response = requests.post(url='https://api.pdfrest.com/upload',
                    data=upload_image_data,
                    headers={'Content-Type': 'application/octet-stream', 'content-filename': 'image_file.png', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})

print("Upload image response status code: " + str(upload_image_response.status_code))

if upload_pdf_response.ok and upload_image_response.ok:
    upload_pdf_response_json = upload_pdf_response.json()
    print(json.dumps(upload_pdf_response_json, indent = 2))

    upload_image_response_json = upload_image_response.json()
    print(json.dumps(upload_image_response_json, indent = 2))


    uploaded_pdf_id = upload_pdf_response_json['files'][0]['id']
    uploaded_file_id = upload_image_response_json['files'][0]['id']
    added_image_data = { "id" : uploaded_pdf_id, "image_id": uploaded_file_id, "x":0, "y":0, "page":1 }
    print(json.dumps(added_image_data, indent = 2))


    print("Processing file...")
    added_image_response = requests.post(url='https://api.pdfrest.com/pdf-with-added-image',
                        data=json.dumps(added_image_data),
                        headers={'Content-Type': 'application/json', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})



    print("Processing response status code: " + str(added_image_response.status_code))
    if added_image_response.ok:
        added_image_response_json = added_image_response.json()
        print(json.dumps(added_image_response_json, indent = 2))

    else:
        print(added_image_response.text)
else:
    print(upload_pdf_response.text)
    print(upload_image_response.text)

File: PHP/Endpoint Examples/JSON Payload/pdf-with-added-image.php
<?php
require 'vendor/autoload.php'; // Require the autoload file to load Guzzle HTTP client.

use GuzzleHttp\Client; // Import the Guzzle HTTP client namespace.
use GuzzleHttp\Psr7\Request; // Import the PSR-7 Request class.
use GuzzleHttp\Psr7\Utils; // Import the PSR-7 Utils class for working with streams.

// By default, we use the US-based API service. This is the primary endpoint for global use.
$apiUrl = "https://api.pdfrest.com";

/* For GDPR compliance and enhanced performance for European users, you can switch to the EU-based service by uncommenting the URL below.
 * For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
 */
//$apiUrl = "https://eu-api.pdfrest.com";

$upload_pdf_client = new Client(['http_errors' => false]);
$upload_pdf_headers = [
  'api-key' => 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
  'content-filename' => 'filename.pdf',
  'Content-Type' => 'application/octet-stream'
];
$upload_pdf_body = file_get_contents('/path/to/pdf_file');
$upload_pdf_request = new Request('POST', $apiUrl.'/upload', $upload_pdf_headers, $upload_pdf_body);
$upload_pdf_res = $upload_pdf_client->sendAsync($upload_pdf_request)->wait();
echo $upload_pdf_res->getBody() . PHP_EOL;

$upload_pdf_response_json = json_decode($upload_pdf_res->getBody());

$uploaded_pdf_id = $upload_pdf_response_json->{'files'}[0]->{'id'};

echo "PDF file successfully uploaded with an id of: " . $uploaded_pdf_id . PHP_EOL;


$upload_image_client = new Client(['http_errors' => false]);
$upload_image_headers = [
  'api-key' => 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
  'content-filename' => 'image.png',
  'Content-Type' => 'application/octet-stream'
];
$upload_image_body = file_get_contents('/path/to/image_file);
$upload_image_request = new Request('POST', $apiUrl.'/upload', $upload_image_headers, $upload_image_body);
$upload_image_res = $upload_image_client->sendAsync($upload_image_request)->wait();
echo $upload_image_res->getBody() . PHP_EOL;

$upload_image_response_json = json_decode($upload_image_res->getBody());

$uploaded_image_id = $upload_image_response_json->{'files'}[0]->{'id'};

echo "Image file successfully uploaded with an id of: " . $uploaded_image_id . PHP_EOL;



$add_image_client = new Client(['http_errors' => false]);
$add_image_headers = [
  'api-key' => 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
  'Content-Type' => 'application/json'
];
$add_image_body = '{"id":"'.$uploaded_pdf_id.'", "image_id": "'.$uploaded_image_id.'", "x":0, "y":0, "page":1}';
$add_image_request = new Request('POST', $apiUrl.'/pdf-with-added-image', $add_image_headers, $add_image_body);
$add_image_res = $add_image_client->sendAsync($add_image_request)->wait();
echo $add_image_res->getBody() . PHP_EOL;
